src/EventLoop.php

```diff
@@ -63,7 +63,7 @@ public function delay(int $milliseconds): Promise;
     /**
      * Creates a deferred, allowing to create and resolve your own promises.
      */
-    public function deferred(): Deferred;
+    public function deferred(callable $canceller = null): Deferred;
```

Comment:
* Missing comment about `$canceller`
* Could be renamed as `$cancelCallback`? 🤔 
* Should be **mandatory**, by design we want to ensure that every function cares about cancellation